src/components/Modal/ExtensionStatusModal.tsx

```diff
@@ -0,0 +1,203 @@
+import React, { useRef } from 'react';
+import styles from './modal.module.scss';
+import { useGetSelfExtensionRequestsQuery } from '@/app/services/tasksApi';
+import { SmallSpinner } from '../tasks/card/SmallSpinner';
+
+type ExtensionStatusModalProps = {
+    isOpen: boolean;
+    onClose: () => void;
+    taskId: string;
+    dev: boolean;
+    assignee: string;
+};
+
+export const ExtensionStatusModal: React.FC<ExtensionStatusModalProps> = ({
+    isOpen,
+    onClose,
+    taskId,
+    dev,
+    assignee,
+}) => {
+    const { data, isLoading, error } = useGetSelfExtensionRequestsQuery(
+        { taskId, dev },
+        { skip: !isOpen }
+    );
+    const modalRef = useRef<HTMLDivElement>(null);
+
+    const handleOutsideClick = (e: React.MouseEvent) => {
+        if (modalRef.current && !modalRef.current.contains(e.target as Node))
+            onClose();
+    };
+
+    const formatDate = (timestamp: number) => {
+        const timestampMs =
+            timestamp > 9999999999 ? timestamp : timestamp * 1000;
+        const date = new Date(timestampMs);
+        return `${
+            date.getMonth() + 1
+        }/${date.getDate()}/${date.getFullYear()}, ${date.toLocaleTimeString()}`;
+    };
+
+    const getStatusClass = (status: string) =>
+        status === 'APPROVED'
+            ? styles.extensionApproved
+            : styles.extensionPending;
```

Comment:
- please move this above the component for the reasons explained in the comment above.